api/instancemutater/machine.go

```diff
@@ -12,16 +12,21 @@ import (
 	"github.com/juju/juju/api/base"
 	apiwatcher "github.com/juju/juju/api/watcher"
 	"github.com/juju/juju/apiserver/params"
+	"github.com/juju/juju/core/instance"
 	"github.com/juju/juju/core/status"
 	"github.com/juju/juju/core/watcher"
 )
 
 //go:generate mockgen -package mocks -destination mocks/caller_mock.go github.com/juju/juju/api/base APICaller,FacadeCaller
 //go:generate mockgen -package mocks -destination mocks/machinemutater_mock.go github.com/juju/juju/api/instancemutater MutaterMachine
 type MutaterMachine interface {
+
+	// InstanceId returns the provider specific instance id for this machine
+	InstanceId() (string, error)
+
 	// CharmProfilingInfo returns info to update lxd profiles on the machine
 	// based on the given unit names.
```

Comment:
"based on the given unit names." can be removed based on changes to CharmProfilingInfo()